test/e2e/storage/testsuites/provisioning.go

```diff
@@ -690,3 +710,54 @@ func prepareDataSourceForProvisioning(
 
 	return dataSourceRef, cleanupFunc
 }
+
+func preparePVCDataSourceForProvisioning(
+	node framework.NodeSelection,
+	client clientset.Interface,
+	dynamicClient dynamic.Interface,
+	source *v1.PersistentVolumeClaim,
+	class *storagev1.StorageClass,
+) (*v1.TypedLocalObjectReference, func()) {
+	var err error
+	if class != nil {
+		ginkgo.By("[Initialize dataSource]creating a StorageClass " + class.Name)
+		_, err = client.StorageV1().StorageClasses().Create(class)
+		framework.ExpectNoError(err)
+	}
+
+	ginkgo.By("[Initialize dataSource]creating a source PVC")
+	sourcePVC, err := client.CoreV1().PersistentVolumeClaims(source.Namespace).Create(source)
+	framework.ExpectNoError(err)
+
+	wantStatus := v1.ClaimBound
+	if *class.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+		wantStatus = v1.ClaimPending
```

Comment:
Unsure what is the purpose of checking for state if it's just going to be pending.  Since the purpose of this function is not to test PVC binding, what if we just don't check for PVC Phase and just wait for the pod to succeed to populate the data?  If there actually is some binding failure, it will show up as pod failed to run.

Reply:
Well, as you pointed out the code before was creating a PVC and waiting for "bound" status, as a result if one were to pass in a local-disk for example or anything using "FirstConsumer" binding, then this helper function in the test suite would fail.

So by adding the check above we "know" not to fail if the test is using a WaitForFirstConsumer PVC, and we don't fail  here because it never went to Bound.  That way it's up to the person implementing the test to decide how they want to  deal with things like checking usage against a pod etc, but it makes this suite flexible enough to be used for both.

Reply:
This function is specific to preparing pvc data source, so I don't think it really needs to check pvc bound state. It only cares that the volume was successfully populated with some data.

If you still want to keep the pvc bound check, then I would move the check to after the pod is successful. This is how other tests handle both immediate and delayed binding.

Reply:
That works, I'll just drop it out altogether, thanks.